flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/catalog/CatalogTableITCase.scala

```diff
@@ -0,0 +1,450 @@
+import org.apache.flink.table.api.TableEnvironment
+import org.apache.flink.table.api.scala.{BatchTableEnvironment, StreamTableEnvironment}
+import org.apache.flink.table.factories.utils.TestCollectionTableFactory
+import org.apache.flink.types.Row
+
+import org.junit.Assert.assertEquals
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+import org.junit.{Before, Ignore, Test}
+
+import java.util
+
+import scala.collection.JavaConversions._
+
+/** Test cases for catalog table. */
+@RunWith(classOf[Parameterized])
+class CatalogTableITCase(isStreaming: Boolean) {
+
+  private val batchExec: ExecutionEnvironment = ExecutionEnvironment.getExecutionEnvironment
+  private var batchEnv: BatchTableEnvironment = _
+  private val streamExec: StreamExecutionEnvironment = StreamExecutionEnvironment
+    .getExecutionEnvironment
+  private var streamEnv: StreamTableEnvironment = _
+
+  private val SOURCE_DATA = List(
+      toRow(1, "a"),
+      toRow(2, "b"),
+      toRow(3, "c")
+  )
+
+  private val DIM_DATA = List(
+    toRow(1, "aDim"),
+    toRow(2, "bDim"),
+    toRow(3, "cDim")
+  )
+
+  implicit def rowOrdering: Ordering[Row] = Ordering.by((r : Row) => {
+    val builder = new StringBuilder
+    0 until r.getArity foreach(idx => builder.append(r.getField(idx)))
+    builder.toString()
+  })
+
+  @Before
+  def before(): Unit = {
+    batchExec.setParallelism(4)
+    streamExec.setParallelism(4)
+    batchEnv = BatchTableEnvironment.create(batchExec)
+    streamEnv = StreamTableEnvironment.create(streamExec)
+    TestCollectionTableFactory.reset()
+    TestCollectionTableFactory.isStreaming = isStreaming
+  }
+
+  def toRow(args: Any*):Row = {
+    val row = new Row(args.length)
+    0 until args.length foreach {
+      i => row.setField(i, args(i))
+    }
+    row
+  }
+
+  def tableEnv: TableEnvironment = {
+    if (isStreaming) {
+      streamEnv
+    } else {
+      batchEnv
+    }
+  }
+
+  def execJob(name: String) = {
+    if (isStreaming) {
+      streamExec.execute(name)
+    } else {
+      batchExec.execute(name)
+    }
+  }
+
+  @Test
+  def testInsertInto(): Unit = {
+    val sourceData = List(
+      toRow(1, "1000", 2),
+      toRow(2, "1", 3),
+      toRow(3, "2000", 4),
+      toRow(1, "2", 2),
+      toRow(2, "3000", 3)
+    )
+    TestCollectionTableFactory.initData(sourceData)
+    val sourceDDL =
+      """
+        |create table t1(
+        |  a int,
+        |  b varchar,
+        |  c int
+        |) with (
+        |  connector = 'COLLECTION'
+        |)
+      """.stripMargin
+    val sinkDDL =
+      """
+        |create table t2(
+        |  a int,
+        |  b varchar,
+        |  c int
+        |) with (
+        |  connector = 'COLLECTION'
+        |)
+      """.stripMargin
+    val query =
+      """
+        |insert into t2
+        |select t1.a, t1.b, (t1.a + 1) as c from t1
+      """.stripMargin
+    tableEnv.sql(sourceDDL)
+    tableEnv.sql(sinkDDL)
+    tableEnv.sql(query)
```

Comment:
Can we add some tests that covers multiple statements in one sql with ';' delimiter ? 

Btw, IT cases are very heavy (the community is trying to reduce travis time). I think one for ETL and one for aggregation is enough. We can add some unit tests to verify the plan which is much more lightweight.

Reply:
We do not multiple statements now, i think we do not need plan test now, because this change only affect table source and sink but not the plan, an ITCase is enough, we need an end-to-end test for catalog table now.